packages/api/src/modules/vaults/vaults.service.ts

```diff
@@ -100,6 +105,54 @@ export class VaultsService {
     return await this.readyVaultInDB(createdVault.data);
   }
 
+  private createVaultParams(params: VaultParamsDto): ICredbull.FixedYieldVaultParamsStruct {
+    const baseVaultParams = {
+      asset: params.asset,
+      shareName: params.shareName,
+      shareSymbol: params.shareSymbol,
+      custodian: params.custodian,
+    };
+
+    const contractRoles = {
+      owner: params.owner,
+      operator: params.operator,
+      custodian: params.custodian,
+    };
+
+    const depositWindowParam = {
+      opensAt: params.depositOpensAt,
+      closesAt: params.depositClosesAt,
+    };
+
+    const matureWindowParam = {
+      opensAt: params.redemptionOpensAt,
+      closesAt: params.redemptionClosesAt,
+    };
+
+    const windowVaultParams = {
+      depositWindow: depositWindowParam,
+      matureWindow: matureWindowParam,
+    };
```

Comment:
for the deposit windows, i was thinking we could have a struct to represent the open and close at.  generally, i'd prefer object types that are more descriptive over simple primitives.  

for example below, this also raises the question, shouldn't a window be between two Dates (not numbers).

```    
interface WindowParam {
  opensAt: BigNumber;
  closesAt: BigNumber;
}
```


```    
    const depositWindowParam: WindowParam = {
      opensAt: params.depositOpensAt,
      closesAt: params.depositClosesAt,
    };

    const matureWindowParam: WindowParam = {
      opensAt: params.redemptionOpensAt,
      closesAt: params.redemptionClosesAt,
    };```

Reply:
I love the idea of `interface`s with the param values, then using multiple inheritance to realise a single params. 
Sadly, interfaces don't allow state variables [link](https://docs.soliditylang.org/en/v0.8.26/contracts.html#interfaces) (I would love to be wrong here!)
So we would have to use accessor functions, which would be a lot more cumbersome and would reduce the elegance of the solution I think.
```solidity
// Read-only interface
interface IContractRoles {
    function getOwner() external returns (address);
    function getOperator() external returns (address);
    function getCustodian() external returns (address);
}

struct VaultParameters is IContractRoles {
...
```
Even still, as far as I can tell, only contracts themselves can realise `interface`s, so making this design a non-starter.